// ng-timeline/ng-timeline.es5.js
import { Component, Input, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { animate, keyframes, query, stagger, style, transition, trigger } from '@angular/animations';
var TimelineEntry = /** @class */ (function () {
    function TimelineEntry() {
    }
    return TimelineEntry;
}());
var TimelineEntryBuilder = /** @class */ (function () {
    function TimelineEntryBuilder() {
        this._timelineEntry = new TimelineEntry();
    }
    /**
     * @param {?} title
     * @return {?}
     */
    TimelineEntryBuilder.prototype.title = function (title) {
        this._timelineEntry.title = title;
        return this;
    };
    /**
     * @param {?} body
     * @return {?}
     */
    TimelineEntryBuilder.prototype.body = function (body) {
        this._timelineEntry.body = body;
        return this;
    };
    /**
     * @param {?} major
     * @return {?}
     */
    TimelineEntryBuilder.prototype.major = function (major) {
        this._timelineEntry.major = major;
        return this;
    };
    /**
     * @return {?}
     */
    TimelineEntryBuilder.prototype.build = function () {
        return this._timelineEntry;
    };
    return TimelineEntryBuilder;
}());
var NgTimelineComponent = /** @class */ (function () {
    function NgTimelineComponent() {
    }
    return NgTimelineComponent;
}());
NgTimelineComponent.decorators = [
    { type: Component, args: [{
                selector: 'ng-timeline',
                template: "\n    <div class=\"ng-timeline\">\n      <div class=\"ng-timeline-entries\" [@.disabled]=\"!enableAnimation\" [@timelineAnimation]=\"entries.length\">\n        <div class=\"ng-timeline-entry\" *ngFor=\"let entry of entries; index as i\">\n          <div class=\"ng-timeline-title\" [ngClass]=\"{'big': entry.major }\" [innerHtml]=\"entry.title\"></div>\n          <div class=\"ng-timeline-body\" [innerHtml]=\"entry.body\"></div>\n        </div>\n      </div>\n    </div>\n  ",
                animations: [
                    trigger('timelineAnimation', [
                        transition('* => *', [
                            query(':enter', style({ opacity: 0 }), { optional: true }),
                            query(':enter', stagger('200ms', [
                                animate('1s ease-out', keyframes([
                                    style({ opacity: 0, transform: 'translateY(-75%)', offset: 0 }),
                                    style({ opacity: .5, transform: 'translateY(-25px)', offset: 0.5 }),
                                    style({ opacity: 1, transform: 'translateY(0)', offset: 1.0 }),
                                ]))
                            ]), { optional: true })
                        ])
                    ])
                ]
            },] },
];
/**
 * @nocollapse
 */
NgTimelineComponent.ctorParameters = function () { return []; };
NgTimelineComponent.propDecorators = {
    'entries': [{ type: Input },],
    'enableAnimation': [{ type: Input },],
};
var NgTimelineModule = /** @class */ (function () {
    function NgTimelineModule() {
    }
    return NgTimelineModule;
}());
NgTimelineModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    BrowserAnimationsModule
                ],
                declarations: [
                    NgTimelineComponent,
                ],
                exports: [
                    NgTimelineComponent,
                ]
            },] },
];
/**
 * @nocollapse
 */
NgTimelineModule.ctorParameters = function () { return []; };
/**
 * Generated bundle index. Do not edit.
 */
export { TimelineEntry, TimelineEntryBuilder, NgTimelineModule, NgTimelineComponent as ɵa };
