// ng-timeline/bundles/ng-timeline.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/platform-browser/animations'), require('@angular/animations')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/common', '@angular/platform-browser/animations', '@angular/animations'], factory) :
	(factory((global['ng-timeline'] = {}),global.ng.core,global.ng.common,global.ng.platformBrowser.animations,global.ng.animations));
}(this, (function (exports,core,common,animations,animations$1) { 'use strict';

var TimelineEntry = /** @class */ (function () {
    function TimelineEntry() {
    }
    return TimelineEntry;
}());
var TimelineEntryBuilder = /** @class */ (function () {
    function TimelineEntryBuilder() {
        this._timelineEntry = new TimelineEntry();
    }
    /**
     * @param {?} title
     * @return {?}
     */
    TimelineEntryBuilder.prototype.title = function (title) {
        this._timelineEntry.title = title;
        return this;
    };
    /**
     * @param {?} body
     * @return {?}
     */
    TimelineEntryBuilder.prototype.body = function (body) {
        this._timelineEntry.body = body;
        return this;
    };
    /**
     * @param {?} major
     * @return {?}
     */
    TimelineEntryBuilder.prototype.major = function (major) {
        this._timelineEntry.major = major;
        return this;
    };
    /**
     * @return {?}
     */
    TimelineEntryBuilder.prototype.build = function () {
        return this._timelineEntry;
    };
    return TimelineEntryBuilder;
}());
var NgTimelineComponent = /** @class */ (function () {
    function NgTimelineComponent() {
    }
    return NgTimelineComponent;
}());
NgTimelineComponent.decorators = [
    { type: core.Component, args: [{
                selector: 'ng-timeline',
                template: "\n    <div class=\"ng-timeline\">\n      <div class=\"ng-timeline-entries\" [@.disabled]=\"!enableAnimation\" [@timelineAnimation]=\"entries.length\">\n        <div class=\"ng-timeline-entry\" *ngFor=\"let entry of entries; index as i\">\n          <div class=\"ng-timeline-title\" [ngClass]=\"{'big': entry.major }\" [innerHtml]=\"entry.title\"></div>\n          <div class=\"ng-timeline-body\" [innerHtml]=\"entry.body\"></div>\n        </div>\n      </div>\n    </div>\n  ",
                animations: [
                    animations$1.trigger('timelineAnimation', [
                        animations$1.transition('* => *', [
                            animations$1.query(':enter', animations$1.style({ opacity: 0 }), { optional: true }),
                            animations$1.query(':enter', animations$1.stagger('200ms', [
                                animations$1.animate('1s ease-out', animations$1.keyframes([
                                    animations$1.style({ opacity: 0, transform: 'translateY(-75%)', offset: 0 }),
                                    animations$1.style({ opacity: .5, transform: 'translateY(-25px)', offset: 0.5 }),
                                    animations$1.style({ opacity: 1, transform: 'translateY(0)', offset: 1.0 }),
                                ]))
                            ]), { optional: true })
                        ])
                    ])
                ]
            },] },
];
/**
 * @nocollapse
 */
NgTimelineComponent.ctorParameters = function () { return []; };
NgTimelineComponent.propDecorators = {
    'entries': [{ type: core.Input },],
    'enableAnimation': [{ type: core.Input },],
};
var NgTimelineModule = /** @class */ (function () {
    function NgTimelineModule() {
    }
    return NgTimelineModule;
}());
NgTimelineModule.decorators = [
    { type: core.NgModule, args: [{
                imports: [
                    common.CommonModule,
                    animations.BrowserAnimationsModule
                ],
                declarations: [
                    NgTimelineComponent,
                ],
                exports: [
                    NgTimelineComponent,
                ]
            },] },
];
/**
 * @nocollapse
 */
NgTimelineModule.ctorParameters = function () { return []; };

exports.TimelineEntry = TimelineEntry;
exports.TimelineEntryBuilder = TimelineEntryBuilder;
exports.NgTimelineModule = NgTimelineModule;
exports.ɵa = NgTimelineComponent;

Object.defineProperty(exports, '__esModule', { value: true });

})));
